modules/rest/src/main/java/org/apache/ignite/internal/rest/cluster/ClusterManagementController.java

```diff
@@ -84,16 +113,31 @@ public CompletableFuture<Void> init(@Body InitCommand initCommand) {
         });
     }
 
-    private static ClusterState mapClusterState(org.apache.ignite.internal.cluster.management.ClusterState clusterState) {
+    private ClusterState mapClusterState(org.apache.ignite.internal.cluster.management.ClusterState clusterState) {
         return new ClusterState(
                 clusterState.cmgNodes(),
                 clusterState.metaStorageNodes(),
                 clusterState.igniteVersion().toString(),
                 new ClusterTag(clusterState.clusterTag().clusterName(), clusterState.clusterTag().clusterId()),
-                clusterState.formerClusterIds()
+                clusterState.formerClusterIds(),
+                mapClusterStatus(clusterState)
         );
     }
 
+    private ClusterStatus mapClusterStatus(org.apache.ignite.internal.cluster.management.ClusterState clusterState) {
+        Set<String> metaStorageNodes = clusterState.metaStorageNodes();
+        long presentMetaStorageNodes = topologyService.allMembers().stream()
+                .map(ClusterNode::name)
+                .filter(metaStorageNodes::contains)
+                .count();
+
+        if (presentMetaStorageNodes <= metaStorageNodes.size() / 2) {
+            return ClusterStatus.MS_MAJORITY_LOST;
```

Comment:
The fact that we have enough nodes in the physical topology does not necessarily imply we have an MG majority. For instance, something on levels higher than the network might prevent a leader to be elected. The most accurate way would be to try executing a read command against the MS group (like reading its current revision) and handle a timeout.